router/pkg/watcher/watcher_test.go

```diff
@@ -0,0 +1,88 @@
+package watcher_test
+
+import (
+	"context"
+	"errors"
+	"github.com/stretchr/testify/require"
+	"github.com/wundergraph/cosmo/router/pkg/watcher"
+	"go.uber.org/zap"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+)
+
+var waitForEvents = 500 * time.Millisecond
+
+func getEvent(event <-chan []watcher.Event) ([]watcher.Event, error) {
+	select {
+	case events := <-event:
+		return events, nil
+	case <-time.After(1 * time.Second):
+		return nil, errors.New("timed out while waiting for watcher events")
+	}
+}
+
+func TestChange(t *testing.T) {
+	dir, err := os.MkdirTemp("", "watcher-tests")
+	require.NoError(t, err)
+	tempFile := filepath.Join(dir, "config.json")
+	defer os.RemoveAll(dir)
+
+	err = os.WriteFile(tempFile, []byte("a"), 0644)
+	require.NoError(t, err)
+
+	ctx := context.Background()
+	eventCh := make(chan []watcher.Event)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	w, err := watcher.NewWatcher(zap.NewNop())
+	require.NoError(t, err)
+	err = w.Watch(ctx, tempFile, func(events []watcher.Event) error {
+		select {
+		case eventCh <- events:
+		case <-ctx.Done():
+		}
+		return nil
+	})
+	require.NoError(t, err)
+
+	time.Sleep(waitForEvents)
```

Comment:
you can use require.Eventually

Reply:
I can use it. It makes the test a bit quicker.

Reply:
Done